### PR TITLE
[14.0][MIG] stock_request_mrp: Migration to 14.0

### PR DESCRIPTION
**Pull request by pedrobaeza**
@Tecnativa TT39761

### CODE REVIEW

**Comment by pedrobaeza** [comment]
The tests error seems not related to this module.

**Comment by rousseldenis** [comment]
> The tests error seems not related to this module.

It seems so:

image

@pedrobaeza Could you check ?

File: stock_request_mrp/models/mrp_production.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    @api.depends("stock_request_ids")
+    def _compute_stock_request_ids(self):
+        for rec in self:
+            rec.stock_request_count = len(rec.stock_request_ids)
+
+    def action_view_stock_request(self):
+        """
+        :return dict: dictionary value for created view
+        """
+        action = self.env.ref("stock_request.action_stock_request_form").read()[0]
```
Comment (rousseldenis) [inline stock_request_mrp/models/mrp_production.py]: Please use _for_xml_id()

File: stock_request_mrp/models/stock_request.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ):
+            raise ValidationError(
+                _(
+                    "You have linked to a Manufacture Order "
+                    "that belongs to another company."
+                )
+            )
+
+    def action_view_mrp_production(self):
+        action = self.env.ref("mrp.mrp_production_action").read()[0]
```
Comment (rousseldenis) [inline stock_request_mrp/models/stock_request.py]: same as above

File: stock_request_mrp/models/stock_request_order.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    )
+
+    @api.depends("stock_request_ids")
+    def _compute_production_ids(self):
+        for req in self:
+            req.production_ids = req.stock_request_ids.mapped("production_ids")
+            req.production_count = len(req.production_ids)
+
+    def action_view_mrp_production(self):
+        action = self.env.ref("mrp.mrp_production_action").read()[0]
```
Comment (rousseldenis) [inline stock_request_mrp/models/stock_request_order.py]: Same here

File: stock_request_mrp/tests/test_stock_request_mrp.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
+# Copyright 2016-20 ForgeFlow S.L. (https://www.forgeflow.com)
+# Copyright 2022 Tecnativa - Pedro M. Baeza
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from odoo import fields
+from odoo.tests import Form, common
+
+
+class TestStockRequestMrp(common.TransactionCase):
+    def setUp(self):
```
Comment (rousseldenis) [inline stock_request_mrp/tests/test_stock_request_mrp.py:13]: You maybe want to use setUpClass instead

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Review by victoralmau — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@rousseldenis can you please check again?

**Comment by pedrobaeza** [comment]
@rousseldenis please update your review

**Review by rousseldenis — APPROVED** [review]
LGTM

**Pull request merged**